rollup-interface/src/node/services/da.rs

```diff
@@ -30,17 +32,33 @@ pub trait DaService: Send + Sync + 'static {
     >;
 
     /// The error type for fallible methods.
-    type Error: fmt::Debug + Send + Sync + Display;
-
-    /// Retrieve the data for the given height, waiting for it to be
-    /// finalized if necessary. The block, once returned, must not be reverted
-    /// without a consensus violation.
-    async fn get_finalized_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error>;
+    type Error: core::fmt::Debug + Send + Sync + core::fmt::Display;
 
     /// Fetch the block at the given height, waiting for one to be mined if necessary.
-    /// The returned block may not be final, and can be reverted without a consensus violation
+    /// The returned block may not be final, and can be reverted without a consensus violation.
+    /// Call it for the same height are allowed to return different results.
+    /// Should always returns the block at that height on the best fork.
     async fn get_block_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error>;
 
+    /// Fetch the [`DaSpec::BlockHeader`] of the last finalized block.
+    async fn get_last_finalized_block_header(
+        &self,
+    ) -> Result<<Self::Spec as DaSpec>::BlockHeader, Self::Error>;
+
+    /// Subscribe to finalized headers as they are finalized.
+    /// Optimized version of `get_last_finalized_block_header`.
+    /// For the environments that support [`tokio::sync::broadcast`].
+    fn subscribe_finalized_header(
+        &self,
+    ) -> Result<tokio::sync::broadcast::Receiver<<Self::Spec as DaSpec>::BlockHeader>, Self::Error>;
```

Comment:
Why do we return a `Receiver` instead of an associated type that implements `Stream`?

Reply:
I think `Stream` is a better option.

Originally I followed [`LedgerRpcProvider::subscribe_slots`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/nightly/rollup-interface/src/node/rpc/mod.rs#L305)